heptio/sonobuoy
# Add Details map to post-processed Items

**What this PR does / why we need it**:
Currently just used for junit and specifically meant to address
a gap in necessary capabilities for processing e2e results.

When tests fail, you want to see what tests fail but also why they
failed. Without this functionality, you still have to resort to
untarring the entire tarball just to see basic logs like "timeout"

**Which issue(s) this PR fixes**
Fixes #827

**Special notes for your reviewer**:

To test this out with a failure I used:
```
sonobuoy run --sonobuoy-image=schnake/sonobuoy:stdoutResultsInfo --plugin e2e --e2e-focus="\[Driver: local\]\[LocalVolumeType: dir-link-bindmounted\] \[Testpattern: Pre-provisioned PV \(default fs\)\] subPath should support existing directories when readOnly specified in the volumeSource" --wait --image-pull-policy=Always --wait-output=spinner && outfile=$(sonobuoy retrieve) && rm -rf results && mkdir results && tar -xf $outfile -C results && code results && sonobuoy delete --wait
```

Then
```
sonobuoy results $outfile --mode detailed|jq 'select(has("details"))'
```

An interesting note is that this works:
```
sonobuoy results $outfile --mode detailed|jq 'select(has("details"))' | jq .details.failure
```

But not:
```
sonobuoy results $outfile --mode detailed|jq 'select(has("details"))' | jq .details.system-out
```

I'm not sure if it is my shell or jq or just a really bad idea to have the hyphenated key. I can modify that if needed to just strip non [a-zA-Z] values but I'd prefer to just keep the junit key names.

Update: See https://github.com/stedolan/jq/issues/38
You can do it but need to use `.["system-out"]` but then also need to worry about your shell escaping so mine looked awkward: `jq 'select(has("details"))|.details'|jq .\[\"failure\"\]`

**Release note**:
```
Adds a new `Details` field to the `results.Item` type which allows the post-processing to capture test output and report on it.
```
